clang/lib/CIR/CodeGen/CIRGenExprCXX.cpp

```diff
@@ -1572,10 +1572,15 @@ RValue CIRGenFunction::emitCXXDestructorCall(GlobalDecl Dtor,
   commonBuildCXXMemberOrOperatorCall(*this, DtorDecl, This, ImplicitParam,
                                      ImplicitParamTy, CE, Args, nullptr);
   assert((CE || Dtor.getDecl()) && "expected source location provider");
+
+  auto cxxDtor = cir::CXXDtorAttr::get(
+      &getMLIRContext(),
+      getContext().getRecordType(DtorDecl->getParent()).getAsString());
```

Comment:
IMO instead of counting on clang's to-string mechanism, I wonder if we could store a better way of identifying the kind.  This is I believe going to include the template arguments along the way, which are going to be confusing.

IF what we are really looking for is something like "this is a `std::` collection", I wonder if we should do the work to only do this in the case of `DtorDecl->isInStdNamespace`, and perhaps slightly more work to see if the base 'name' is correct (that is, only if `DtorDecl->getParent()->getName()` -is-in a certain list).